sup3r/utilities/utilities.py

```diff
@@ -20,14 +19,42 @@
 from packaging import version
 from scipy import ndimage as nd
 from scipy.interpolate import RegularGridInterpolator, interp1d
-from scipy.ndimage import zoom
-from scipy.ndimage import gaussian_filter
+from scipy.ndimage import gaussian_filter, zoom
 
 np.random.seed(42)
 
 logger = logging.getLogger(__name__)
 
 
+class Timer:
+    """Timer class for timing and storing function call times."""
+
+    def __init__(self):
+        self.log = {}
+
+    def __call__(self, fun, *args, **kwargs):
+        """Time function call and store elapsed time in self.log.
+
+        Parameters
+        ----------
+        fun : function
+            Function to time
+        *args : list
+            positional arguments for fun
+        **kwargs : dict
+            keyword arguments for fun
+
+        Returns
+        -------
+        output of fun
+        """
+        t0 = time.time()
+        out = fun(*args, **kwargs)
+        t_elap = time.time() - t0
+        self.log[f'elapsed:{fun.__name__}'] = t_elap
+        return out
```

Comment:
Why don't you want `Timer` to be part of `TensorboardMixIn`?

How does the log discern consecutive calls to the function? Just because you write the log to tensorboard every epoch? 

I guess i could see how Timer would be useful outside of the tensorboard stuff with additional instance functions. Maybe an auto-logger? You don't need to add now but could be useful in the future.

Reply:
Yeah I figured it was a pretty general method and yeah log is written for each batch actually.

Reply:
makes sense 👍